src/Makefile

```diff
@@ -51,232 +41,30 @@ TARGET_SOURCES := accesslogger.cpp \
-	@sort $(COVERAGE_MASTER_LIST) | comm -23 $(COVERAGE_LIST_TMP) - > $(COVERAGE_LIST)
-	@if grep -q ^ $(COVERAGE_LIST) ; then \
-		echo "Error: some files unexpectedly have less than 100% code coverage:" ; \
-		cat $(COVERAGE_LIST) ; \
-		/bin/false ; \
-		echo "Marking build unstable." ; \
-	fi
-
-# Get quick coverage data at the command line. Add --branches to get branch info
-# instead of line info in report.  *.gcov files generated in current directory
-# if you need to see full detail.
-.PHONY: coverage_raw
-coverage_raw: | $(TEST_OUT_DIR)
-	$(GCOVR) $(COVERAGEFLAGS) --keep
-
-.PHONY: debug
-debug: build_test
-	gdb --args $(TARGET_BIN_TEST) $(EXTRA_TEST_ARGS)
-
-# Don't run VG against death tests; they don't play nicely.
-# Be aware that running this will count towards coverage.
-# Don't send output to console, or it might be confused with the full
-# unit-test run earlier.
-# Test failure should not lead to build failure - instead we observe
-# test failure from Jenkins.
-.PHONY: vg
-vg: build_test | $(TEST_OUT_DIR)
-	-valgrind --xml=yes --xml-file=$(VG_XML) $(VGFLAGS) \
-	  $(TARGET_BIN_TEST) --gtest_filter='-*DeathTest*' $(EXTRA_TEST_ARGS) > $(VG_OUT) 2>&1
-
-# Check whether there were any errors from valgrind. Output to screen any errors found,
-# and details of where to find the full logs.
-# The output file will contain <error><kind>ERROR</kind></error>, or 'XPath set is empty'
-# if there are no errors.
-.PHONY: vg-check
-vg-check: vg
-	@xmllint --xpath '//error/kind' $(VG_XML) 2>&1 | \
-		sed -e 's#<kind>##g' | \
-		sed -e 's#</kind>#\n#g' | \
-		sort > $(VG_LIST)
-	@if grep -q -v "XPath set is empty" $(VG_LIST) ; then \
-		echo "Error: some memory errors have been detected" ; \
-		cat $(VG_LIST) ; \
-		echo "See $(VG_XML) for further details." ; \
-	fi
-
-.PHONY: vg_raw
-vg_raw: build_test | $(TEST_OUT_DIR)
-	-valgrind --gen-suppressions=all $(VGFLAGS) \
-	  $(TARGET_BIN_TEST) --gtest_filter='-*DeathTest*' $(EXTRA_TEST_ARGS)
-
-.PHONY: distclean
-distclean: clean
-
-${ROOT}/usr/include/ralf_alarmdefinition.h : ${BUILD_DIR}/bin/alarm_header ${ROOT}/ralf.root/usr/share/clearwater/infrastructure/alarms/ralf_alarms.json
-	${BUILD_DIR}/bin/alarm_header -j "${ROOT}/ralf.root/usr/share/clearwater/infrastructure/alarms/ralf_alarms.json" -n "ralf"
-	mv ralf_alarmdefinition.h $@
-
-# Build rules for GMock/GTest library.
-$(OBJ_DIR_TEST)/gtest-all.o : $(GTEST_SRCS_)
-	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) -I$(GTEST_DIR)/include -I$(GMOCK_DIR) -I$(GMOCK_DIR)/include \
-            -c $(GTEST_DIR)/src/gtest-all.cc -o $@
-
-$(OBJ_DIR_TEST)/gmock-all.o : $(GMOCK_SRCS_)
-	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) -I$(GTEST_DIR)/include -I$(GMOCK_DIR) -I$(GMOCK_DIR)/include \
-            -c $(GMOCK_DIR)/src/gmock-all.cc -o $@
-
-# Build rule for our interposer.
-$(OBJ_DIR_TEST)/test_interposer.so: ut/test_interposer.cpp ut/test_interposer.hpp
-	$(CXX) $(CPPFLAGS) -shared -fPIC -ldl $< -o $@
+ralf_SOURCES := ${COMMON_SOURCES} main.cpp
```

Comment:
I don't think so, I've (effectively) reserved `<target-in-lowercase>_<UPPERCASE>` for communication between `Makefile` and `cpp.mk`, it would be weird to have special exemptions to this rule.